### DISCUSSION TITLE
How to stop VM that was started by a run

### DISCUSSION THREAD

**Discussion by mhanberg** [cirruslabs/cirrus-cli#650] [Q&A]
I am using cirrus cli on apple silicon in conjunction with a GitHub actions self hosted runner. 

I've noticed that if a GitHub actions workflow gets cancelled, a cirrus task that is executing won't get shutdown properly, so the VM seems to just hang around. 

Then, on a subsequent workflow, it will fail because the 2 VM limit has already been reached. 

The GHA runner has the ability to run a script before or after a job, so I am hoping I can run a script to ensure the VM has shutdown. 

Is this possible? I think the problem might be knowing which VM to ensure is shut down, so not to shutdown the wrong own.

**Comment by fkorotkov** [comment]
Do you see a `cirrus` process of a Cirrus CLI instance? I winder if GHA Runner is not terminating the process properly.

**Reply by mhanberg** [reply]
Sorry about the delayed response!

I will have to try bumping up to two runners again and see. 

But yes that's what I assumed. It might be a side affect if GitHub hosted runners are always run in a fresh vm, so it doesn't matter about orphaning processes cuz the whole VM gets torn down.
